pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/utils/DateTimeUtils.java

```diff
@@ -34,8 +34,8 @@ public class DateTimeUtils {
   private DateTimeUtils() {
   }
 
```

Comment:
What about other strings that used to be accepted by older formatter ? 
e.g. 
```
"2020-13-01" 
"2020-12-01 "
"2020-12-01 12"
"2020-12-01 12:00"
```

Reply:
Hm that's a good point, we might just need to roll back to `SimpleDateFormat` in order to maintain backward compatibility and either allocate an SDF instance per call or use thread locals as suggested by @gortiz.

Reply:
`"2020-12-01 "` can be handled by introducing a trim and `"2020-12-01 12"` / `"2020-12-01 12:00"` could be handled by also making the `mm` / `ss` parts optional in the `DateTimeFormatter` pattern. While SDF is able to parse `"2020-13-01"`, it's actually pretty non-standard behavior because it gets parsed into `2021-01-01` so I think it might be fine to drop support for that?